### PR TITLE
Block users from importing argo workflows in pipeline import modal

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: [RHOAIENG-8569](https://issues.redhat.com/browse/RHOAIENG-8569)

## Description
Argo workflow is not supported with pipelinev2, so blocking the user to import it, and adding error message on for pipeline detail pages.

<details><summary>When user try to import argo workflow. </summary>

Screenshot 2024-07-11 at 2 59 37 PM
</details>

<details><summary>When user try to upload new pipeline version with argo workflow. </summary>

Screenshot 2024-07-16 at 6 10 36 PM
</details>

<details><summary>Disable tooltip for "Create run" and "Create schedule"</summary>

Screenshot 2024-07-11 at 4 56 54 PM
Screenshot 2024-07-11 at 4 57 37 PM
</details>

<details><summary>Pipeline details page</summary>

image
</details>

<details><summary>Pipeline run details page</summary>

Screenshot 2024-07-16 at 6 14 24 PM

</details>

<details><summary>Pipeline schedule details page</summary>

image
</details>

## How Has This Been Tested?
1. Try importing a argo workflow (YAML attached in JIRA)
2. If you already have argo workflow added:
a. Try creating a run/schedule with it.
b. Check the error on the detail pages.

## Test Impact
Added unit and cypress tests

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`
@yannnz

### CODE REVIEW

**Comment by dpanshug** [comment]
WIP: needs UX review for the error and tooltip messages

**Comment by dpanshug** [comment]
@yannnz @kaedward Can you pls review the screenshots attached.

File: frontend/src/concepts/pipelines/content/import/PipelineImportModal.tsx [hunk 1]
```diff
@@ -47,13 +48,22 @@ const PipelineImportModal: React.FC<PipelineImportModalProps> = ({ isOpen, onClo
...
-        .then((pipeline) => onBeforeClose(pipeline))
-        .catch((e) => {
-          setImporting(false);
-          setError(e);
-        });
+      if (isArgoWorkflow) {
+        setImporting(false);
+        setError(
+          new Error(
+            'The selected pipeline was created using the Kubeflow v1 SDK, which is not supported by this UI. Select a pipeline that was created or recompiled using the Kubeflow v2 SDK.',
```
Comment (manaswinidas) [inline frontend/src/concepts/pipelines/content/import/PipelineImportModal.tsx:55]: Can this be used as a constant? This is used in two different modals.

File: frontend/src/concepts/pipelines/content/import/PipelineImportModal.tsx [hunk 2]
```diff
@@ -152,7 +162,12 @@ const PipelineImportModal: React.FC<PipelineImportModalProps> = ({ isOpen, onClo
           </StackItem>
           {error && (
             <StackItem>
-              <Alert title="Error creating pipeline" isInline variant="danger">
+              <Alert
+                data-testid="import-modal-error"
+                title={isArgoWorkflow ? 'Unsupported pipeline version' : 'Error creating pipeline'}
```
Comment (manaswinidas) [inline frontend/src/concepts/pipelines/content/import/PipelineImportModal.tsx:167]: same here. I see this used in 3 different areas excluding tests

File: frontend/src/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTableRow.tsx [hunk 3]
```diff
@@ -90,6 +97,10 @@ const PipelineVersionTableRow: React.FC<PipelineVersionTableRowProps> = ({
                   ),
                 );
               },
+              isAriaDisabled: isCreateDisabled,
+              tooltipProps: isCreateDisabled
+                ? { content: 'Cannot create schedule for Kubeflow v1 SDK pipelines' }
```
Comment (manaswinidas) [inline frontend/src/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTableRow.tsx:102]: same here.

File: frontend/src/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTableRow.tsx [hunk 4]
```diff
@@ -78,6 +81,10 @@ const PipelineVersionTableRow: React.FC<PipelineVersionTableRowProps> = ({
                   ),
                 );
               },
+              isAriaDisabled: isCreateDisabled,
+              tooltipProps: isCreateDisabled
+                ? { content: 'Cannot create run for Kubeflow v1 SDK pipelines' }
```
Comment (manaswinidas) [inline frontend/src/concepts/pipelines/content/tables/pipelineVersion/PipelineVersionTableRow.tsx:86]: same here. If this is a const, we don't have to worry about typos and making the same change in different strings.

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/concepts/pipelines/content/import/utils.ts [hunk 5]
```diff
@@ -10,3 +10,10 @@ export enum PipelineUploadOption {
   URL_IMPORT,
   FILE_UPLOAD,
 }
+
+// Utility function to extract Kind from Pipeline YAML
+export const extractKindFromPipelineYAML = (yamlFile: string): string | undefined => {
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/import/utils.ts:15]: why not parse the yaml and then check if it is kind workflow instead of having this custom parsing logic

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx [hunk 6]
```diff
@@ -3,39 +3,66 @@ import { Stack, StackItem, Tooltip } from '@patternfly/react-core';
...
+import { isArgoWorkflow } from '~/concepts/pipelines/content/tables/utils';
 
 type PipelineSelectorTableRowProps = {
   obj: PipelineKFv2 | PipelineVersionKFv2 | ExperimentKFv2;
+  isSchedule?: boolean;
+  isCreatePage?: boolean;
   onClick: () => void;
 };
 
-const PipelineSelectorTableRow: React.FC<PipelineSelectorTableRowProps> = ({ obj, onClick }) => {
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:12]: This component deals with experiments, pipelines, and versions... however you are adding version specific logic to this component. All that logic should be moved up a level to PipelineVersionSelector. Then you can pass the row a isDisabled field and additionally a isDisabledReason.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx [hunk 7]
```diff
@@ -91,23 +95,11 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
     <>
       <ApplicationsPage
         breadcrumb={
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx:97]: why did you need to create a bunch of new components here?

File: frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx [hunk 8]
```diff
@@ -247,21 +247,27 @@ export const usePipelineTaskTopology = (
   executions?: Execution[] | null,
   events?: Event[] | null,
   artifacts?: Artifact[],
-): PipelineNodeModelExpanded[] =>
+): PipelineNodeModelExpanded[] | Error =>
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx:250]: why are you returning an error here?

You should instead follow our previous ways of handling state and return [data, loaded, error] if we need that

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx [hunk 9]
```diff
@@ -51,41 +53,45 @@ const PipelineDetailsActions: React.FC<PipelineDetailsActionsProps> = ({
           <DropdownItem key="upload-version" onClick={() => setIsVersionImportModalOpen(true)}>
             Upload new version
           </DropdownItem>,
-          <DropdownSeparator key="separator-create" />,
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx:54]: in some places you disable the create button and others like this area you hide? Can we make it consistent and make them disabled instead

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx [hunk 10]
```diff
@@ -3,39 +3,66 @@ import { Stack, StackItem, Tooltip } from '@patternfly/react-core';
...
+          )
         }
         triggerRef={tooltipRef}
       />
-      <Tr ref={tooltipRef} onRowClick={onClick} isClickable>
+      <Tr
+        ref={tooltipRef}
+        onRowClick={!(isCreatePage && isDisabled) ? onClick : undefined}
+        isClickable={!(isCreatePage && isDisabled)}
+      >
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:65]: this does not have the disabled styling

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]
You are also able to still get the invalid version selected in the create form if you create a run from view runs section of the invalid pipeline.

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts [hunk 11]
```diff
@@ -597,6 +598,24 @@ describe('Pipelines', () => {
     pipelineImportModal.findSubmitButton().should('be.enabled');
   });
 
+  it('imports fails with Argo workflow', () => {
```
Comment (Gkrumbach07) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts:601]: you are missing test coverage on other areas that were changed

File: frontend/src/concepts/pipelines/apiHooks/useLatestPipelineVersion.ts [hunk 12]
```diff
@@ -27,10 +28,9 @@ export const useLatestPipelineVersion = (
     const response = await api.listPipelineVersions({}, pipelineId, {
       sortField: 'created_at',
       sortDirection: 'desc',
-      pageSize: 1,
     });
 
-    return response.pipeline_versions?.[0] || null;
+    return response.pipeline_versions?.find((p) => !isArgoWorkflow(p.pipeline_spec)) || null;
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/concepts/pipelines/apiHooks/useLatestPipelineVersion.ts:33]: If we don't modify this hook, we will only get the latest version, and finding the supported version will not be possible. Do you have a better alternative to modify this?

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx (hunk 7 above)
Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx:97]: Yeah, this is unnecessary now. Initially, I created a new ApplicationsPage component for unsupported pipelines and was reusing this breadcrumb component. However, I rolled back to use the existing ApplicationsPage. This new component is not required anymore. Do you suggest I undo this change or keep it as is because the breadcrumb component is little bigger and having separate file is fine.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx (hunk 9 above)
Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx:54]: I hide them instead of disabled, because that was done before in this component. But we can disable that also, let me change this.

File: frontend/src/concepts/pipelines/apiHooks/useLatestPipelineVersion.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/apiHooks/useLatestPipelineVersion.ts:33]: thats fine, if the latest version is invalid then we should handler that after.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx (hunk 7 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx:97]: i would undo it

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx (hunk 9 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx:54]: we hid them becuase a valid reason. we should disable so we can tell them why

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx (hunk 10 above)
Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:65]: Had a [conversation](https://patternfly.slack.com/archives/C4FM977N0/p1721295556345339) with Patternfly team. Currently, there is no method to apply disabled styling on clickable rows. I'll raise a feature request with Patternfly. Until then, we can use `isStriped` if @yannnz agrees??

Disabled rows with `isStriped`
Screenshot 2024-07-19 at 1 48 32 PM

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:65]: i dont think we should use isStriped. it doesn't seem like it looks disabled still. maybe its better to just hide them if we cant disable properly

Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:65]: @yannnz wdyt?

Comment (yannnz) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow.tsx:65]: +1 to Gage. I think hiding them is a good approach. We use the same strategy for Archived experiment under the Experiment dropdown list under the create Run page.  @dpanshug

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts (hunk 11 above)
Comment (dpanshug) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts:601]: added few more test cases, let me if any thing i've not covered

File: frontend/src/__mocks__/mockPipelineVersionsProxy.ts [hunk 13]
```diff
@@ -1,13 +1,23 @@
...
 } from '~/concepts/pipelines/kfTypes';
 
+export type ArgoWorkflowVersion = Omit<PipelineVersionKFv2, 'pipeline_spec'> & {
+  pipeline_spec: K8sResourceCommon;
+};
+
+export type BuildMockPipelinveVersionsType = PipelineKFCallCommon<{
+  pipeline_versions: (PipelineVersionKFv2 | ArgoWorkflowVersion)[];
+}>;
+
```
Comment (dpanshug) [inline frontend/src/__mocks__/mockPipelineVersionsProxy.ts:20]: Need to create these types to accommodate argo workflow. Is it fine to add it here, as these are only required in mocks and tests?

File: frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx [hunk 14]
```diff
@@ -241,27 +241,49 @@ const getNodesForTasks = (
...
+        root: {
+          dag: { tasks },
+          inputDefinitions,
+        },
+      } = pipelineSpec);
+    } catch (e) {
+      setNodes([]);
+      setLoaded(true);
+      setError(new Error('Error processing pipelineSpec'));
+      return;
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx:285]:
```
components,
        deploymentSpec: { executors },
        root: {
          dag: { tasks },
          inputDefinitions,
        },
      } = pipelineSpec
```
This will throw the error with argo workflow, as it doesn't have inputDefinitions and other.

File: frontend/src/__mocks__/mockPipelineVersionsProxy.ts (hunk 13 above)
Comment (Gkrumbach07) [reply frontend/src/__mocks__/mockPipelineVersionsProxy.ts:20]: `ArgoWorkflowVersion` can exists in the types for pipelines

File: frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx [hunk 15]
```diff
@@ -51,6 +52,12 @@ const PipelineVersionSelector: React.FC<PipelineVersionSelectorProps> = ({
     data: versions,
   } = usePipelineVersionSelector(pipelineId);
 
+  const supportedVersions = React.useMemo(
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx:55]: this is fine to do for the create page, but the pipeline version selector on the details page can have the invalid version selected but it is not visible in the dropdown. you might need to add a condition to the version selector for if it should filter out invalid versions

Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx:58]: Made that change, now i'm stuck with other error.

https://github.com/user-attachments/assets/13845e9a-76ed-4b12-acd5-7148cbe4b3aa

When changing the pipeline version, we see this error screen. The issue originates from [this](https://github.com/opendatahub-io/odh-dashboard/blob/f2ce42bedb9a0badd58ef8ec1774a00166a88d05/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx#L262) line, where the code is accessing the `name` property from the pipeline spec.

This error might be due to an improper state update of the `topologyError` variable. I've tried debugging but haven't found a solution yet. Your input would be appreciated.

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx:58]: ill pull the changes and see what i can do

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx:58]: it think its getting too complicated with he topology hook sending back an error, just test if the version is invalid from the get go and use that instead of the topology error. you can probably revert the topoloigy hook to just catching the error and not retuning a bunch of error and loaded states

Here is a simple diff
```diff
diff --git a/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx b/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
index d5e0ceeb..dd155052 100644
--- a/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
+++ b/frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx
@@ -54,24 +54,18 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
     usePipelineVersionById(pipelineId, pipelineVersionId);
   const [pipeline, isPipelineLoaded, pipelineLoadError] = usePipelineById(pipelineId);
 
-  const {
-    nodes,
-    loaded: isTopologyLoaded,
-    error: topologyError,
-  } = usePipelineTaskTopology(pipelineVersion?.pipeline_spec);
+  const { nodes } = usePipelineTaskTopology(pipelineVersion?.pipeline_spec);
+
+  const isInvalidPipelineVersion = isArgoWorkflow(pipelineVersion?.pipeline_spec);
 
   const selectedNode = React.useMemo(() => {
-    if (topologyError) {
+    if (isInvalidPipelineVersion) {
       return null;
     }
     return nodes.find((n) => n.id === selectedId);
-  }, [topologyError, selectedId, nodes]);
+  }, [isInvalidPipelineVersion, selectedId, nodes]);
 
-  const isLoaded =
-    isPipelineVersionLoaded &&
-    isPipelineLoaded &&
-    !!pipelineVersion?.pipeline_spec &&
-    isTopologyLoaded;
+  const isLoaded = isPipelineVersionLoaded && isPipelineLoaded && !!pipelineVersion?.pipeline_spec;
 
   if (pipelineVersionLoadError || pipelineLoadError) {
     const title = 'Pipeline version not found';
@@ -171,7 +165,7 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
           )
         }
       >
-        {topologyError ? (
+        {isInvalidPipelineVersion ? (
           <PipelineNotSupported />
         ) : (
           <PageSection
```

Comment (dpanshug) [reply frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx:58]: thank you it's working fine now, reverted back the changes in usePipelineTaskTopology

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx [hunk 16]
```diff
@@ -55,6 +61,8 @@ const PipelineDetailsActions: React.FC<PipelineDetailsActionsProps> = ({
           </DropdownItem>,
           <DropdownSeparator key="separator-create" />,
           <DropdownItem
+            isDisabled={!isPipelineSupported}
+            tooltip={PIPELINE_CREATE_RUN_TOOLTIP_ARGO_ERROR}
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetailsActions.tsx:65]: this tooltip will not show. you have to use `aria-disabled` instead of isDisabled

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]
You are also need to disable the create run/schedule from the pipeline version table row action dropdown when it is invalid

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm
